addons/survey/controllers/main.py

```diff
@@ -710,14 +710,31 @@ def _generate_report(self, user_input, download=True):
             ('Content-Disposition', report_content_disposition),
         ])
 
-    def _get_user_input_domain(self, survey, line_filter_domain, **post):
+    def _get_user_input_domain(self, survey, suggested_answer_ids, filter_domain, **post):
+        user_input_ids = []
         user_input_domain = ['&', ('test_entry', '=', False), ('survey_id', '=', survey.id)]
-        if line_filter_domain:
-            matching_line_ids = request.env['survey.user_input.line'].sudo().search(line_filter_domain).ids
+
+        # Get the user input line ids where the user answered the suggested answers present in the filters
+        if suggested_answer_ids:
+            for answer_id in suggested_answer_ids:
+                result = set(request.env['survey.user_input.line'].sudo().search([('suggested_answer_id', '=', answer_id)]).user_input_id.ids)
+                user_input_ids.append(result)
+
+        # Get the user input line ids matching the filter domain
+        if filter_domain:
```

Comment:
Hello ! @dbeguin 

The problem here is that we can't really do a big fat domain or at least I don't really see how.

The problem is that the `user_input_line` model is representing a specific question answer so searching on a specific char_box answer will work, on a specific matrix answer will also work, but both at the same time (by using `answer_domain1 AND answer_domain2` in the domain) will always return an empty recordset. So what I was doing here was getting **for each answer_filter** the matching `user_input` and then intersecting everything to only retrieve the `user_input` matching all the filters at the same time.
I tried to think of a way to filter directly on the `user_input` model instead of the `user_input_line` model but I don't think it is possible to retrieve all the `user_input` which `user_input_line_ids` field (= a recordset of `user_input_line` representing all his answers) is matching every filter in a single search.

Moreover the `('suggested_answer_id', 'in', suggested_answer_ids)` part of the domain won't work because the `in` will retrieve all the answers matching at least one of the filters but not all at the same time. For example if `suggested_answer_ids = [17,18]` it will return the users who answered `17 OR 18 OR both` but here we only want the `both` part of the equation.

I tried to explain as best as I could, don't hesitate to tell me if it's still unclear. :)